lib/src/patterns.dart

```diff
@@ -124,3 +124,73 @@ final tablePattern = RegExp(
 /// A pattern which should never be used. It just satisfies non-nullability of
 /// pattern fields.
 final dummyPattern = RegExp('');
+
+/// A [String] pattern to match a named tag like `<table>` or `</table>`.
+const _namedTagDefinition =
+    // Opening tag begins.
+    '<'
+
+    // Tag name.
+    '[a-z][a-z0-9-]*'
+
+    // Attribute begins, see
+    // https://spec.commonmark.org/0.30/#attribute.
+    r'(?:\s+'
+
+    // Attribute name, see
+    // https://spec.commonmark.org/0.30/#attribute-name.
+    '[a-z_:][a-z0-9._:-]*'
+
+    //
+    '(?:'
+    // Attribute value specification, see
+    // https://spec.commonmark.org/0.30/#attribute-value-specification.
+    r'\s*=\s*'
+
+    // Attribute value, see
+    // https://spec.commonmark.org/0.30/#unquoted-attribute-value.
+    r'''(?:[^\s"'=<>`]+?|'[^']*?'|"[^"]*?")'''
+
+    // Attribute ends.
+    ')?)*'
+
+    // Opening tag ends.
+    r'\s*/?>'
+
+    //
+    '|'
+
+    // Closing tag, see
+    // https://spec.commonmark.org/0.30/#closing-tag.
+    r'</[a-z][a-z0-9-]*\s*>';
+
+final htmlBlockPattern = RegExp(
+    '^ {0,3}(?:'
+    '<(?<condition_1>pre|script|style|textarea)'
+    r'(?:\s|>|$)'
+    '|'
+    '(?<condition_2><!--)'
+    '|'
+    r'(?<condition_3><\?)'
+    '|'
+    '(?<condition_4><![a-z])'
+    '|'
+    r'(?<condition_5><!\[CDATA\[)'
+    '|'
+    '</?(?<condition_6>address|article|aside|base|basefont|blockquote|body|'
+    'caption|center|col|colgroup|dd|details|dialog|dir|DIV|dl|dt|fieldset|'
+    'figcaption|figure|footer|form|frame|frameset|h1|h2|h3|h4|h5|h6|head|'
+    'header|hr|html|iframe|legend|li|link|main|menu|menuitem|nav|noframes|ol|'
+    'optgroup|option|p|param|section|source|summary|table|tbody|td|tfoot|th|'
+    'thead|title|tr|track|ul)'
+    r'(?:\s|>|/>|$)'
+    '|'
+
+    // Here is more restricted than the commonmark definition(Rule #7).
```

Comment:
Grammar, I think something like "Here we are more restrictive than the Commonmark definition (Rule #7)."

Reply:
Fixed